HunnitLog/HunnitLog/View/Home/ScoreRow.swift

```diff
@@ -0,0 +1,118 @@
+        static let sliderMaxValue: CGFloat = 5
+        
+        static let badImageName: String = "imgTextboxBad"
+        static let notBadImageName: String = "imgTextboxNotbad"
+        static let sosoImageName: String = "imgTextboxSoso"
+        static let greatImageName: String = "imgTextboxGreat"
+        static let excellentImageName: String = "imgTextboxExcellent"
+    }
+    
+    
+    var body: some View {
+        let roundedValue = round(sliderValue/step)*step
+        
+        HStack(spacing: 7){
+            VStack{
+                HStack{
+                    Circle()
+                        .fill(CustomColor.yellow)
+                        .frame(width: Constants.circleRadius, height: Constants.circleRadius)
+                        .padding(.trailing, Constants.trailing)
+                        
+                    VStack{
+                        HStack{
+                            Text(title)
+                                .font(Constants.titleFont)
+                                Spacer()
+                        }
+                        HStack{
+                            Text("달성률 \(achievement)%")
+                                .font(Constants.achievementFont)
+                                .foregroundColor(CustomColor.coolGray)
+                            Spacer()
+                        }
+                    }
+                }
+                
+                GeometryReader { geometry in
+                    VStack {
+                        HStack{
+                            
+                            switch roundedValue {
+                            case -5 ... -4:
+                                Image(Constants.badImageName)
+                                    .frame(width: CGFloat(geometry.size.width / 10 * CGFloat(5.9 + sliderValue)), height: 32, alignment: .bottomTrailing)
+                            case -3 ... -2:
+                                Image(Constants.notBadImageName)
+                                    .frame(width: CGFloat(geometry.size.width / 10 * CGFloat(5.9 + sliderValue)), height: 32, alignment: .bottomTrailing)
+                            case -1 ... 1:
+                                Image(Constants.sosoImageName)
+                                    .frame(width: CGFloat(geometry.size.width / 10 * CGFloat(5.9 + sliderValue)), height: 32, alignment: .bottomTrailing)
+                            case 2 ... 3:
+                                Image(Constants.greatImageName)
+                                    .frame(width: CGFloat(geometry.size.width / 10 * CGFloat(5.9 + sliderValue)), height: 32, alignment: .bottomTrailing)
+                            default:
+                                Image(Constants.excellentImageName)
+                                    .frame(width: CGFloat(geometry.size.width / 10 * CGFloat(5.9 + sliderValue)), height: 32, alignment: .bottomTrailing)
+                            }
```

Comment:
중복되는 코드를 줄이기 위해 이런식으로 image만 따로 빼내서 계산하고, frame은 공통으로 적용하는 것은 어떨까요? 
```suggestion
func scoreImage(roundedValue: Double) -> Image {
    switch roundedValue {
    case -5 ... -4:
        return Image(Constants.badImageName)
    case -3 ... -2:
        return Image(Constants.notBadImageName)
    case -1 ... 1:
        return Image(Constants.sosoImageName)
    case 2 ... 3:
        return Image(Constants.greatImageName)
    default:
        return Image(Constants.excellentImageName)
    }
}

scoreImage(roundedValue: roundedValue)
    .frame(width: CGFloat(geometry.size.width / 10 * CGFloat(5.9 + sliderValue)),
           height: 32,
           alignment: .bottomTrailing)
```

Reply:
여기 한단계를 2씩 가지지 않고 정수로 나뉘도록 보정해서 각 숫자 하나가 하나의 감정을 의미하도록 사용해보는건 어떨까요?

Reply:
좀 오바같긴한데...

요런느낌으로 enum 정의하고

```swift
    enum Emotion: Int, CaseIterable, Identifiable {
        case bad = 0
        case notbad = 1
        case soso = 2
        case great = 3
        case excelecnt = 4
        
        var image: Image? {
            Image(self.imageName)
        }
        
        var id: Int {
            self.rawValue
        }
        
        var exposureValue: Int {
            switch self {
            case .bad:
                return -5
            case .notbad:
                return -3
            case .soso:
                return 0
            case .great:
                return 3
            case .excelecnt:
                return 5
            }
        }
        
        private var imageName: String {
            switch self {
            case .bad:
                return "imgTextboxBad"
            case .notbad:
                return "imgTextboxNotbad"
            case .soso:
                return "imgTextboxSoso"
            case .great:
                return "imgTextboxGreat"
            case .excelecnt:
                return "imgTextboxExcellent"
            }
            
        }
    }

```

이렇게 사용해볼 수 있을거같아요
```swift
                GeometryReader { geometry in
                    VStack {
                        HStack{
                            Emotion(rawValue: Int(roundedValue))?
                                .image
                                .frame(width: CGFloat(geometry.size.width / 10 * CGFloat(5.9 + sliderValue)), height: 32, alignment: .bottomTrailing)
                            Spacer()
                        }
                        Slider(value: $sliderValue, in: -5...5)
                        .accentColor(CustomColor.lightGray)
                    }
                }
                .frame(height: 70)
                HStack {
                    ForEach(Emotion.allCases) { emotion in
                        Text("\(emotion.exposureValue)")
                        if emotion != .excelecnt {
                            Spacer()
                        }
                    }
                }
```

Reply:
요렇게 코드 량이 좀 줄어들 수 있을 듯 합니다! 
<img width="1414" alt="image" src="https://user-images.githubusercontent.com/11674980/101241904-9529a800-373d-11eb-9ae3-b195bafaae81.png">

Reply:
Commit [7e85fef](https://github.com/hunnit-log/iOS/pull/28/commits/7e85fefc5165963aa8259892beafe8cc94ac5961) 에 해당 내용 수정하여 작성했습니다! 감사합니다!